packages/plugin-link/src/Link.ts

```diff
@@ -132,23 +138,41 @@ export class Link<T extends JWPluginConfig = JWPluginConfig> extends JWPlugin<T>
             link.modifiers.get(Attributes).set('target', params.target);
         }
         return this.editor.execCommand<Char>('insertText', {
-            text: params.label || link.url,
+            text: label || link.url,
             formats: new Modifiers(link),
-            select: true,
             context: params.context,
         });
     }
     unlink(params: LinkParams): void {
         const range = params.context.range;
-        const node = range.start.previousSibling() || range.start.nextSibling();
-        if (!node.is(InlineNode)) return;
 
-        const link = node.modifiers.find(LinkFormat);
-        if (!link) return;
+        if (range.isCollapsed()) {
+            // if no range is selected and we are in a link : remove the complete link.
+            const previousNode = range.start.previousSibling();
+            const nextNode = range.start.nextSibling();
+            const node = Link.isLink(previousNode) ? previousNode : nextNode;
+            if (!Link.isLink(node)) return;
+
+            const link = node.modifiers.find(LinkFormat);
+            const sameLink: Typeguard<InlineNode> = Link.isLink.bind(Link, link);
+            this._removeLinkOnNodes([node, ...node.adjacents(sameLink)]);
+        } else {
+            // if a range is selected : remove any link on the selected range
+            const selectedNodes = range.selectedNodes(InlineNode);
+            // remove the format 'LinkFormat' for all nodes.
+            this._removeLinkOnNodes(selectedNodes);
+        }
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
 
-        const sameLink: Typeguard<InlineNode> = Link.isLink.bind(Link, link);
-        for (const inline of [node, ...node.adjacents(sameLink)]) {
-            inline.modifiers.remove(link);
+    _removeLinkOnNodes(nodes: VNode[]): void {
+        // remove all link modifiers on the provided nodes
```

Comment:
Can I ask you to capitalize the first letters of your comments and add a dot `.` at the end so it looks like a proper sentence ? We all do that (except Christophe when he forgets) and it makes the comments look more professional. This obviously applies to all comments.

Reply:
will do